flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/ScalarOperatorsTest.scala

```diff
@@ -406,5 +406,26 @@ class ScalarOperatorsTest extends ScalarOperatorsTestBase {
       "((((true) === true) || false).cast(STRING) + 'X ').trim",
       "trueX")
     testTableApi(12.isNull, "12.isNull", "false")
+
```

Comment:
I think we can use a separate test method for `Between`. Besides, some validation tests can be added to `org.apache.flink.table.expressions.validation.ScalarOperatorsValidationTest`.